src/main/java/org/folio/mr/service/impl/MediatedRequestActionsServiceImpl.java

```diff
@@ -69,22 +68,35 @@ private void createRequest(MediatedRequestEntity mediatedRequest) {
       mediatedRequest.setConfirmedRequestId(UUID.fromString(request.getId()));
     } else {
       EcsTlr ecsTlr = ecsRequestService.create(mediatedRequest);
-      mediatedRequest.setConfirmedRequestId(UUID.fromString(ecsTlr.getPrimaryRequestId()));
+      Request request = updatePrimaryRequest(mediatedRequest, ecsTlr);
+      updateMediatedRequest(mediatedRequest, ecsTlr, request);
     }
   }
 
+  private Request updatePrimaryRequest(MediatedRequestEntity mediatedRequest, EcsTlr ecsTlr) {
+    Request request = circulationRequestService.get(ecsTlr.getPrimaryRequestId());
+    request.setRequesterId(mediatedRequest.getRequesterId().toString());
```

Comment:
can it be null?

Reply:
No, 
<column name="requester_id" type="uuid">
                <constraints nullable="false"/>
            </column>